File: NavigationSampleSafeJava/app/src/main/java/com/websarva/wings/android/navigationsamplesafejava/ui/MemoListViewModel.java
package com.websarva.wings.android.navigationsamplesafejava.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.websarva.wings.android.navigationsamplesafejava.data.model.Memo;
import com.websarva.wings.android.navigationsamplesafejava.data.repository.MemoRepository;

import java.util.List;

public class MemoListViewModel extends ViewModel {
	private final MemoRepository _memoRepository;

	public MemoListViewModel() {
		_memoRepository = new MemoRepository();
	}

	public LiveData<List<Memo>> getMemoListLiveData() {
		return _memoRepository.prepareMemoListLiveData();
	}
}

File: NavigationSampleSafeKotlin/app/src/main/java/com/websarva/wings/android/navigationsamplesafekotlin/ui/MemoListViewModel.kt
package com.websarva.wings.android.navigationsamplesafekotlin.ui

import androidx.lifecycle.ViewModel
import com.websarva.wings.android.navigationsamplesafekotlin.data.model.Memo
import com.websarva.wings.android.navigationsamplesafekotlin.data.repository.MemoRepository
import kotlinx.coroutines.flow.Flow

class MemoListViewModel : ViewModel() {
	private val _memoRepository = MemoRepository()

	fun getMemoListFlow(): Flow<List<Memo>> {
		return _memoRepository.prepareMemoListFlow()
	}
}
